graph/mutators/comment.js

```diff
@@ -219,20 +220,62 @@ const addCommentTag = ({user, loaders: {Comments}}, {id, tag}) => {
 
 /**
  * Removes a tag from a Comment
- * @param {String} id          identifier of the comment  (uuid)
+ * @param {String} id      identifier of the comment  (uuid)
  * @param {String} tag     name of the tag
  */
 const removeCommentTag = ({user, loaders: {Comments}}, {id, tag}) => {
   return CommentsService.removeTag(id, tag);
 };
 
+/**
+ * Edit a Comment
+ * @param {String} id         identifier of the comment  (uuid)
+ * @param {Object} edit       describes how to edit the comment
+ * @param {String} edit.body  the new Comment body
+ */
+const editComment = async ({user, loaders: {Comments}}, {id, edit}) => {
+  const {body} = edit;
+  const comment = await CommentsService.findById(id);
```

Comment:
Sure! You would have to adjust the mutation for editing a comment to include the asset id in the query params, that would ensure that it was possible to retrieve the data you need. It would update as so:

```js
const lastValidExpiryDate = new Date((new Date()).getTime() - EDIT_WINDOW_MS);
db.comments.update({
  id,
  asset_id,
  author_id: user.id,
  created_at: {
    $gt: lastValidExpiryDate
  }
}, {
  $set: {
    body,
    status,
  },
  $push: {
    body_history: {
      body,
      created_at: new Date()
    },
    status_history: {
      type: status,
      created_at: new Date(),
    }
  }
});
```